### PR TITLE
Refactor Rot[X/Y/Z/11]Gate into [CZ/X/Y/Z]PowGate

### PR DESCRIPTION
**Pull request by Strilanc**
- `__init__` now takes exponent instead of rads/degs/half_turns
- exposed property is exponent instead of half_turns

Part of https://github.com/quantumlib/Cirq/issues/865

### CODE REVIEW

File: cirq/ops/__init__.py [hunk 1]
```diff
@@ -26,10 +26,10 @@
     measure,
     measure_each,
     MeasurementGate,
-    Rot11Gate,
-    RotXGate,
-    RotYGate,
-    RotZGate,
+    CZPowGate,
```
Comment (dabacon) [inline cirq/ops/__init__.py]: we've been using alphabetical and this breaks that style though I agree it's nice to have these near each other

File: cirq/__init__.py [hunk 2]
```diff
@@ -134,10 +134,10 @@
     QubitOrder,
     QubitOrderOrList,
     ReversibleCompositeGate,
-    Rot11Gate,
-    RotXGate,
-    RotYGate,
-    RotZGate,
+    CZPowGate,
```
Comment (dabacon) [inline cirq/__init__.py]: This breaks alphabetical

File: cirq/ops/__init__.py (hunk 1 above)
Comment (Strilanc) [reply cirq/ops/__init__.py]: Done.

File: cirq/__init__.py (hunk 2 above)
Comment (Strilanc) [reply cirq/__init__.py]: Done.

**Pull request merged**